### PR TITLE
Look at again and revise

### PR DESCRIPTION
**Pull request by rocky**

### CODE REVIEW

File: docs/extending/developing-code/code-overview/evaluation.rst [hunk 1]
```diff
@@ -36,37 +38,35 @@ that, the method handle ``b`` needs to be computed. And this comes
...
 
 Function Name to Python method lookup
--------------------------------------
+=====================================
 
 .. index:: Symbol, Predefined, Builtin, Expression
 
 When an ``Expression`` has not been rewritten, the entire function
 invocation in Mathics comes from the first leaf (or ``Head[]``) which
 should be a ``Symbol``. In Python this will be a class some sort, such
```
Comment (mmatera) [inline docs/extending/developing-code/code-overview/evaluation.rst]: Head[] could also be an expression, like in `Derivative[1,2][f][x,y]`. In this case,
`Head[Derivative[1,2][f][x,y]]` is `Derivative[1,2][f]` and
`Head[Head[Derivative[1,2][f][x,y]]]` is `Derivative[1,2]`.
Of course, the typical expressions have Symbols as heads.

File: docs/extending/developing-code/code-overview/evaluation.rst [hunk 2]
```diff
@@ -36,37 +38,35 @@ that, the method handle ``b`` needs to be computed. And this comes
...
-as ``Builtin`` or ``Predefined`` or ``SympyFunction``. These classes
-are described in a later section.
+as ``Builtin`` or ``Predefined`` or ``SympyFunction`` or a method
```
Comment (mmatera) [inline docs/extending/developing-code/code-overview/evaluation.rst]: Also, it is possible to reduce to applying certain replacement rules stored in the Definitions object, which is the most typical case.

File: docs/extending/developing-code/code-overview/evaluation.rst [hunk 3]
```diff
@@ -36,37 +38,35 @@ that, the method handle ``b`` needs to be computed. And this comes
...
 
 The remaining leaves of the ``Expression`` are the parameters to give
 to an ``apply`` method.
 
-In the simplest case, the ``evaluate()`` method is called. This is
+In the simplest case, the *evaluate()* method is called. This is
 used when a function has no parameters or arguments. In other words,
 it looks like a constant or variable name, and usually is prefaced
-with a ``$`. Examples here are ``$VersionNumber`` or ``$MachineName``.
+with a ``$``. Examples here are ``$VersionNumber`` or ``$MachineName``.
```
Comment (mmatera) [inline docs/extending/developing-code/code-overview/evaluation.rst]: So, the thumb rule for choosing the base class will be:
* To define Builtin Symbols that represent variables, use `Predefined`
* To define Symbols that represent heads of functions that delegate the evaluation on Sympy, use SympyFunction
* For general symbols that represent heads of functions, which can be evaluated both by applying replacement rules
of by calling a Python function over their arguments, 
* To define symbols corresponding to atomic expressions, use  `AtomBuiltin`.

File: docs/extending/developing-code/code-overview/evaluation.rst (hunk 1 above)
Comment (rocky) [reply docs/extending/developing-code/code-overview/evaluation.rst]: Do I have this correct that if the head is an expression then after it is evaluated the result should be a function which is then applied to the remaining leaves? 

If so, I'll need to rework this more.

File: docs/extending/developing-code/code-overview/evaluation.rst (hunk 2 above)
Comment (rocky) [reply docs/extending/developing-code/code-overview/evaluation.rst]: Ok - I missed that. This is addressed in c42e721.

File: docs/extending/developing-code/code-overview/evaluation.rst (hunk 3 above)
Comment (rocky) [reply docs/extending/developing-code/code-overview/evaluation.rst]: I have expanded this and put it in the section which enumerates Objects. This information while useful doesn't really belong here which is trying just how we handle function evaluation at a high level without trying to get into too many details.  (And already there is too much detail here making it longer and harder to get a high-level idea.)

**Comment by rocky** [comment]
Merging as this is better than what was there. We'll come back and hit it again later.

**Pull request merged**